jcl/src/java.base/share/classes/java/lang/Throwable.java

```diff
@@ -591,18 +591,19 @@ private void readObject(ObjectInputStream s)
 		return null;
 	}
 	for (int i=0; i < stack.length - duplicates; i++) {
+		StackTraceElement element = stack[i];
 		if (!outOfMemory) {
 			try {
-				appendTo(err, "\tat " + stack[i], indents);			 //$NON-NLS-1$
+				appendTo(err, "\tat " + element, indents);			 //$NON-NLS-1$
 			} catch(OutOfMemoryError e) {
 				outOfMemory = true;
 			}
 		}
 		if (outOfMemory) {
 			appendTo(err, "\tat ", indents); //$NON-NLS-1$
-			Util.printStackTraceElement(stack[i], null, err, false);
+			appendTo(err, element.toString());
```

Comment:
The point was to avoid toString() because it doesn't work very well when OOM occurs.

Reply:
Worth adding a comment to explain this. All the code uses appendTo() in order to avoid creating objects so for OOM you can still get a stack trace to the console.

Reply:
Above the `Throwable.printStackTrace()` function, there is a comment

```
/* 
 * CMVC 97756 try to continue printing as much as possible of the stack trace even
 *            in the presence of OutOfMemoryErrors.
 */
```

I'm thinking I'll move that into the function doc, maybe:

```
/**
 * Print stack trace.
 *
 * The stack trace is constructed with appendTo() and avoids allocating Objects (i.e. Strings)
 * to try to continue printing as much as possible of the stack trace even in the presence of
 * OutOfMemoryErrors (CMVC 97756).
 *
 * @param	err	
 ...
```

Reply:
I used `toString()` since `StackTraceElement.includeClassLoaderName` is a private field. Is there any restriction on introducing public fields which aren't part of the java specification? If not, I could change `StackTraceElement.includeClassLoaderName` to be public.

Reply:
Public/protected fields/methods cannot be added. If passing a parameter doesn't work, create an internal access via java.lang.VMAccess.